flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/source/MySqlSourceITCase.java

```diff
@@ -264,6 +292,132 @@ public void testConsumingTableWithoutPrimaryKey() {
+
+        String[] snapshotForSingleTable =
+                new String[] {
+                    "+I[101, user_1, Shanghai, 123567891234]",
+                    "+I[102, user_2, Shanghai, 123567891234]",
+                    "+I[103, user_3, Shanghai, 123567891234]",
+                    "+I[109, user_4, Shanghai, 123567891234]",
+                    "+I[110, user_5, Shanghai, 123567891234]",
+                    "+I[111, user_6, Shanghai, 123567891234]",
+                    "+I[118, user_7, Shanghai, 123567891234]",
+                    "+I[121, user_8, Shanghai, 123567891234]",
+                    "+I[123, user_9, Shanghai, 123567891234]",
+                    "+I[1009, user_10, Shanghai, 123567891234]",
+                    "+I[1010, user_11, Shanghai, 123567891234]",
+                    "+I[1011, user_12, Shanghai, 123567891234]",
+                    "+I[1012, user_13, Shanghai, 123567891234]",
+                    "+I[1013, user_14, Shanghai, 123567891234]",
+                    "+I[1014, user_15, Shanghai, 123567891234]",
+                    "+I[1015, user_16, Shanghai, 123567891234]",
+                    "+I[1016, user_17, Shanghai, 123567891234]",
+                    "+I[1017, user_18, Shanghai, 123567891234]",
+                    "+I[1018, user_19, Shanghai, 123567891234]",
+                    "+I[1019, user_20, Shanghai, 123567891234]",
+                    "+I[2000, user_21, Shanghai, 123567891234]"
+                };
+        TypeSerializer<RowData> serializer =
+                source.getTransformation().getOutputType().createSerializer(env.getConfig());
+        String accumulatorName = "dataStreamCollect_" + UUID.randomUUID();
+        CollectSinkOperatorFactory<RowData> factory =
+                new CollectSinkOperatorFactory(serializer, accumulatorName);
+        CollectSinkOperator<RowData> operator = (CollectSinkOperator) factory.getOperator();
+        CollectResultIterator<RowData> iterator =
+                new CollectResultIterator(
+                        operator.getOperatorIdFuture(),
+                        serializer,
+                        accumulatorName,
+                        env.getCheckpointConfig());
+        CollectStreamSink<RowData> sink = new CollectStreamSink(source, factory);
+        sink.name("Data stream collect sink");
+        env.addOperator(sink.getTransformation());
+        JobClient jobClient = env.executeAsync("snapshotSplitTest");
+        iterator.setJobClient(jobClient);
+        JobID jobId = jobClient.getJobID();
+
+        // Trigger failover once some snapshot records has been sent by sleeping source
+        if (iterator.hasNext()) {
+            triggerFailover(
+                    FailoverType.JM,
+                    jobId,
+                    miniClusterResource.getMiniCluster(),
+                    () -> sleepMs(100));
+        }
+
+        List<String> expectedSnapshotData = new ArrayList<>(Arrays.asList(snapshotForSingleTable));
```

Comment:
Rename `snapshotForSingleTable` to `expectedSnapshotData` and remove this line.

Reply:
assertEqualsInAnyOrder accept `List`, I'll improve this piece